src/mercury_engine_data_structures/formats/bmsld.py

```diff
@@ -34,125 +37,186 @@
     ),
 )
 
-Components = {
-    "TRIGGER": Struct(
-        command=StrId,
-        arguments=make_vector(FunctionArgument),
-    ),
-    "SPAWNGROUP": Struct(
-        command=StrId,
-        arguments=make_vector(FunctionArgument),
-    ),
-    "SPAWNPOINT": Struct(
-        command=StrId,
-        arguments=make_vector(FunctionArgument),
-    ),
-    "STARTPOINT": Struct(
-        command=StrId,
-        arguments=make_vector(FunctionArgument),
-    ),
-    "MODELUPDATER": Struct(
-        command=StrId,
-        arguments=make_vector(FunctionArgument),
-    ),
-}
-
 ProperActor = Struct(
-    type=StrId,
-    position=CVector3D,
-    rotation=CVector3D,
-    components=make_vector(
+    "type" / StrId,
+    "position" / CVector3D,
+    "rotation" / CVector3D,
+    "components"
```

Comment:
this should be renamed to `component_functions`, since that's what it's actually a list of. the helper functions you're adding should be renamed to match too